src/azul/deployment.py

```diff
@@ -277,5 +285,149 @@ def _sanitize_tf(tf_config: JSON) -> JSON:
+    cases to make parsing Terraform easier.
+
+    >>> list(_flatten_nested_block({"foo": "bar"}))
+    [('foo', 'bar')]
+
+    >>> list(_flatten_nested_block([{"foo": "bar"}]))
+    [('foo', 'bar')]
+
+    >>> list(_flatten_nested_block({"foo": "bar", "baz": "qux"}))
+    [('foo', 'bar'), ('baz', 'qux')]
+
+    >>> list(_flatten_nested_block([{"foo": "bar"}, {"baz": "qux"}]))
+    [('foo', 'bar'), ('baz', 'qux')]
+
+    >>> list(_flatten_nested_block([{"foo": "bar", "baz": "qux"}]))
+    [('foo', 'bar'), ('baz', 'qux')]
+    """
+    if isinstance(block, dict):
+        return itertools.chain(block.items())
+    else:
+        return itertools.chain.from_iterable(attr.items() for attr in block)
+
+
+def _populate_tags(tf_config: JSON) -> JSON:
+    """
+    Adds tags to all resources and changes the `name` tag to `Name` for tagged
+    AWS resources.
+
+    >>> _populate_tags({"resource": [{"aws_vpc": {"res_name": {"tags": {"name": "foo"}}}}]})
+    {'resource': [{'aws_vpc': {'res_name': {'tags': {'Name': 'foo'}}}}]}
+
+    >>> _populate_tags({"resource": [{"google_service_account": {"res_name": {"tags": {"name": "foo"}}}}]})
+    {'resource': [{'google_service_account': {'res_name': {'tags': {'name': 'foo'}}}}]}
+    """
+    if 'resource' not in tf_config:
+        return tf_config
+    resources = []
+    for resource_type, resource in _flatten_nested_block(tf_config['resource']):
+        for resource_name, properties in _flatten_nested_block(resource):
+            properties = add_tags(resource_type, properties)
+            if 'tags' in properties and 'name' in properties['tags'] and resource_type.startswith('aws_'):
+                properties['tags']['Name'] = properties['tags'].pop('name')
+            resources.append({resource_type: {resource_name: properties}})
+    return {**tf_config, 'resource': resources}
+
+
 def emit_tf(tf_config: Optional[JSON]):
-    return emit(tf_config) if tf_config is None else emit(_sanitize_tf(tf_config))
+    return emit(tf_config) if tf_config is None else emit(_sanitize_tf(_populate_tags(tf_config)))
+
+
+def tags(resource_name: str, **kwargs) -> Dict[str, str]:
+    """
+    Returns tags named for cloud resources based on :class:`azul.Config`.
+    If no `resource_name` is passed, a kwarg `name` is expected.
+
+    :param str resource_name: Value for the `name` tag. By convention, a unique
+                              fully-qualified resource name, unless the resource
+                              being tagged already has a name assigned to it, in
+                              which case that name should be used even if it is
+                              not unique.  See
+                              :meth:`azul.Config.qualified_resource_name`.
+
+    :param kwargs: Provided key-value pairs can be used to specify additional
+                   tags or to override default tags.
+    """
+    component_suffix = config.unqualified_resource_name_or_none(resource_name)[0]
```

Comment:
I prefer unpacking like this instead of tuple indexing.

```suggestion
    resource_name, stage = config.unqualified_resource_name_or_none(resource_name)
```
or else maybe
```suggestion
    component_suffix, _ = config.unqualified_resource_name_or_none(resource_name)
```